tests/conftest.py

```diff
@@ -72,26 +72,37 @@ def polarion_testsuite_properties(record_testsuite_property):
         )
 
 
-def cluster_teardown(log_level="DEBUG"):
+def cluster_teardown(deployer, log_level="DEBUG"):
     log.info("Destroying the test cluster")
-    destroy_cluster(config.ENV_DATA['cluster_path'], log_level)
+    deployer.destroy_cluster(log_level)
     log.info("Destroying the test cluster complete")
 
 
 @pytest.fixture(scope="session", autouse=True)
 def cluster(request, log_cli_level):
```

Comment:
Shouldn't this fixture be much more minimal? My understanding was that all of this logic was going to happen in the deployer's deploy method but it seems like that only executes the openshift-installer command. Other steps in the process like writing the configuration to a file, downloading the client/installer, adding volumes, etc. all still happen in this fixture. 

I believe this fixture should be extremely minimal with pretty much no implementation details of the deployment process so that it can be a general purpose fixture which will work for all deployment types (aws, openstack, vmware, etc.)

Reply:
I almost trimmed this fixture to be very minimal in the new update, please let me know what do you think.